src/app/Scenes/ReverseSearchImageResults/ReverseSearchImageResults.tsx

```diff
@@ -0,0 +1,118 @@
+
+type SearchImageResults = NonNullable<
+  ReverseSearchImageResultsQueryResponse["reverseImageSearch"]
+>["results"]
+
+export const ReverseSearchImageResults = () => {
+  const { showActionSheetWithOptions } = useActionSheet()
+  const [fetching, setFetching] = useState(false)
+  const [results, setResults] = useState<SearchImageResults | null>(null)
+
+  const handleSeachByImage = async () => {
+    try {
+      const images = await showPhotoActionSheet(showActionSheetWithOptions, true, false)
+      const image = images[0]
+      let resizedWidth = image.width
+      let resizedHeight = image.height
+
+      /**
+       * For optimal performance of TinEye, image should be 600px in size in the smallest dimension
+       * For example, image with 1600x1200 size should be resized to 800x600
+       */
+      if (image.width > image.height) {
+        resizedHeight = 600
+      } else {
+        resizedWidth = 600
+      }
+
+      setFetching(true)
+
+      const resizedImage = await resizeImage({
+        uri: image.path,
+        width: resizedWidth,
+        height: resizedHeight,
+        quality: 85,
+        onlyScaleDown: true,
+      })
+      const fileImage = new ReactNativeFile({
+        uri: resizedImage.uri,
+        name: resizedImage.name,
+        type: "image/jpeg",
+      })
+
+      const execute = fetchQuery<ReverseSearchImageResultsQuery>(
+        defaultEnvironment,
+        graphql`
+          query ReverseSearchImageResultsQuery($file: Upload!) {
+            reverseImageSearch(image: $file) {
+              results {
+                ...ReverseSearchImageResultItem_item
+                artwork {
+                  internalID
+                }
+              }
+            }
+          }
+        `,
+        {
+          file: fileImage,
+        }
+      )
+      const response = await execute.toPromise()
+      const imageResults = response?.reverseImageSearch?.results ?? []
+
+      setResults(imageResults)
+    } catch (error) {
+      console.error(error)
+      Alert.alert("Something went wrong", (error as Error).message)
+    } finally {
+      setFetching(false)
+    }
+  }
+
+  const isEmptyResults = Array.isArray(results) && results.length === 0
+  const hasResults = Array.isArray(results) && results.length > 0
+
+  return (
+    <Screen>
+      <Screen.Header onBack={goBack} />
+      <Screen.Body scroll>
+        <Button block loading={fetching} onPress={handleSeachByImage}>
+          Search by image
+        </Button>
+
+        <Box mt={2} mb={1}>
+          {!!isEmptyResults && (
+            <Text textAlign="center" my={2}>
+              No matches
+            </Text>
+          )}
+
+          {!!hasResults &&
+            results.map((result) => (
+              <ReverseSearchImageResultItemFragmentContainer
+                key={result.artwork?.internalID}
+                item={result}
+              />
+            ))}
```

Comment:
and then we can use the `hasResults` to conditionally render the noResults/list of results
```suggestion
          {!!hasResults ?
          (<>
          results.map((result) => (
              <ReverseSearchImageResultItemFragmentContainer
                key={result.artwork?.internalID}
                item={result}
              />
            ))
            </>)
            :
            <Text textAlign="center" my={2}>
              No matches
            </Text>
       }
            
```

Reply:
In this case, we will have "No matches" as soon as the user opened the screen. I wanted to render the search results or "No matches" only after the user selected the image